### PR TITLE
fix(navigation): supply retry function to blocker

### PR DESCRIPTION
**Pull request by civing**
fix Lundalogik/crm-feature#2780
Co-authored-by: Malin Tjärnemo <maltja1@gmail.com>

### CODE REVIEW

File: packages/interfaces/src/service/navigation-service.interface.ts [hunk 1]
```diff
@@ -157,14 +149,23 @@ export type LocationChange = Partial<Location> & {
  */
 export type NavigationEvent = CustomEvent<Location>;
 
+/**
+ * Object describing a navigation from a {Location} to another {Location}.
```
Comment (jgroth) [inline packages/interfaces/src/service/navigation-service.interface.ts:153]:
```suggestion
 * {@link Transition} describes the transition when navigation from a {@link Location} to another {@link Location}.
```

File: packages/interfaces/src/service/navigation-service.interface.ts [hunk 2]
```diff
@@ -157,14 +149,23 @@ export type LocationChange = Partial<Location> & {
...
+ *
+ * @param {Location} to location to navigate to
+ * @param {Location} from current location before navigation
+ * @param {Function} retry function to retry navigation
+ */
+export type Transition = {
+    to: Location;
+    from: Location;
+    retry: Function;
```
Comment (jgroth) [inline packages/interfaces/src/service/navigation-service.interface.ts:162]:
```suggestion
    retry: () => void;
```

File: packages/interfaces/src/service/navigation-service.interface.ts [hunk 3]
```diff
@@ -157,14 +149,23 @@ export type LocationChange = Partial<Location> & {
...
+};
+
 /**
  * Function to determine if navigation should be blocked or not.
  *
- * @param {Location} newLocation location to navigate to
- * @param {Location} currentLocation current location before navigation
+ * @param {Location} transition the transition to evaluate
```
Comment (jgroth) [inline packages/interfaces/src/service/navigation-service.interface.ts:168]:
```suggestion
 * @param {Transition} transition the transition to evaluate
```

**Review by jgroth — COMMENTED** [review]
Looks good, just a few improvement suggestions :slightly_smiling_face:

File: packages/interfaces/src/service/navigation-service.interface.ts [hunk 4]
```diff
@@ -71,8 +71,8 @@ export interface Navigator {
      * If the blocker function returns true navigation is blocked. If it returns
      * false navigation will proceed as normal, if not blocked by other blockers.
      *
-     * To later on resume navigation that was blocked, you may save away the new
-     * location provided to the blocker function.
+     * To later on resume navigation that was blocked, you can call the supplied
+     * retry method.
```
[the remark this answers is not in the document]
Comment (civing) [reply packages/interfaces/src/service/navigation-service.interface.ts]: ⚡

File: packages/interfaces/src/service/navigation-service.interface.ts (hunk 2 above)
Comment (civing) [reply packages/interfaces/src/service/navigation-service.interface.ts]: ⚡

File: packages/interfaces/src/service/navigation-service.interface.ts (hunk 3 above)
Comment (civing) [reply packages/interfaces/src/service/navigation-service.interface.ts]: ⚡

**Review by jgroth — DISMISSED** [review]

**Review by jgroth — APPROVED** [review]

**Pull request merged**